orchestrator/services/celery.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2019-2020 SURF.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from http import HTTPStatus
+from typing import Any, Callable, Dict, List, Optional
+from uuid import UUID
+
+from orchestrator.api.error_handling import raise_status
+from orchestrator.db import ProcessTable
+from orchestrator.services.processes import create_process
+from orchestrator.targets import Target
+from orchestrator.types import State
+from orchestrator.workflows import get_workflow
+
+SYSTEM_USER = "SYSTEM"
+
+
+def _celery_start_process(
+    workflow_key: str, user_inputs: Optional[List[State]], user: str = SYSTEM_USER, **kwargs: Any
+) -> UUID:
+    """Client side call of Celery."""
+    from orchestrator.services.tasks import NEW_TASK, NEW_WORKFLOW, get_celery_task
+
+    workflow = get_workflow(workflow_key)
+    if not workflow:
+        raise_status(HTTPStatus.NOT_FOUND, "Workflow does not exist")
```

Comment:
Raising an HTTP exception from a service function is a bit odd, i think it would be better to raise something else and handle that in the API appropriately

Reply:
I know, but apparently this is how it is done in all the places where orchestrator-core checks for a workflow_key. Probably not going to change this now because for sure it will break tests and increases the scope of this ticket.

Reply:
Ok!